File: lista-de-exercicios/exercicio-12/Somador.test.ts
import { jest } from '@jest/globals';
import { Somador } from './Somador';

describe('Classe Somador', () => {
    test('Deve retornar um novo array com todos os elementos de dois ou mais arrays', () => {
        const somador2 = new Somador([[1], [2]]);

        expect(somador2.somar()).toStrictEqual([1, 2]);

        const somador3 = new Somador([[0, 1], [2], [3]]);

        expect(somador3.somar()).toStrictEqual([0, 1, 2, 3]);

        const somador4 = new Somador([[0, 1], [2], [3], [4, 5, 6]]);

        expect(somador4.somar()).toStrictEqual([0, 1, 2, 3, 4, 5, 6]);
    })
})

File: lista-de-exercicios-java/src/test/java/org/example/exercicio12/SomadorTest.java
package org.example.exercicio12;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class SomadorTest {
    @DisplayName("Deve retornar um novo array com todos os elementos de dois ou mais arrays")
    @Test
    void deveRetornarNovoArrayComElementosDosArrays() {
        Somador somador1 = new Somador(new int[][] {{1}, {2}});
        Somador somador2 = new Somador(new int[][] {{0, 1}, {2}, {3}});
        Somador somador3 = new Somador(new int[][] {{1, 2}, {3, 4}, {5, 6}});

        assertArrayEquals(new int[] {1, 2}, somador1.somar());
        assertArrayEquals(new int[] {0, 1, 2, 3,}, somador2.somar());
        assertArrayEquals(new int[] {1, 2, 3, 4, 5, 6}, somador3.somar());
    }
}